drivers/SmartThings/zigbee-lock/src/init.lua

```diff
@@ -291,6 +293,35 @@ local name_slot = function(driver, device, command)
   end
 end
 
+local migrate = function(driver, device, command)
+  local lock_users = {}
+  local lock_credentails = {}
+  local index = 0
+  for code_slot, code_name in pairs(lock_utils.get_lock_codes(device)) do
+    table.insert(lock_users, {userIndex = index, userType = "guest", userName = code_name})
+    table.insert(lock_credentails, {userIndex = index, credentialIndex = code_slot, crednetialType = "pin"})
+    index = index + 1
+  end
+
+  local code_length  = device:get_field(capabilities.lockCodes.codeLength)
```

Comment:
here you would want to be pulling from the device state store with 

`device:get_latest_state("main", lockCodes.ID, lockCodes.codeLength.NAME)`

`get_latest_state` gives you the most recent value of a capability attribute for a given device